dhis-2/dhis-api/src/main/java/org/hisp/dhis/keyjsonvalue/KeyJsonValueService.java

```diff
@@ -30,26 +30,48 @@
 import java.util.Date;
 import java.util.List;
 
+import org.hisp.dhis.keyjsonvalue.KeyJsonNamespaceProtection.ProtectionType;
+import org.springframework.security.access.AccessDeniedException;
+
 /**
  * @author Stian Sandvold
+ * @author Jan Bernitt
  */
 public interface KeyJsonValueService
 {
     /**
-     * Retrieves a list of existing namespaces.
+     * Applies the configuration for the provided protection so it is considered
+     * by this service in future requests.
      *
-     * @return a list of strings representing the existing namespaces.
+     * @param protection configuration for protection protection
+     */
+    void addProtection( KeyJsonNamespaceProtection protection );
+
+    /**
+     * Removes any {@link KeyJsonNamespaceProtection} configuration for the
+     * given namespace (if exists).
+     *
+     * @param namespace the namespace for which to remove configuration
+     */
+    void removeProtection( String namespace );
+
+    /**
+     * True, if there is at least a single value for the provided namespace.
+     *
+     * @param namespace the namespace to check
+     * @return true, if the namespace exists, else false
      */
-    List<String> getNamespaces( boolean isAdmin );
+    boolean isUsedNamespace( String namespace );
 
     /**
-     * Retrieves list of KeyJsonValue objects belonging to the specified
-     * namespace.
+     * Retrieves a list of existing namespaces.
      *
-     * @param namespace the namespace where the key is associated
-     * @return list of matching KeyJsonValues
+     * This does not include {@link ProtectionType#HIDDEN} namespaces that the
+     * current user cannot see.
+     *
+     * @return a list of strings representing the existing namespaces.
      */
-    List<KeyJsonValue> getKeyJsonValuesInNamespace( String namespace, boolean isAdmin );
```

Comment:
Was only used in context of deleting namespaces. This now happens within the service so no further external access is needed.